### PR TITLE
feat: extended verification lib, verified info popover

### PR DESCRIPTION
**Pull request by martines3000**
- [x] verified info popover on hover
- [x] verify lib

### CODE REVIEW

File: libs/extended-verification/src/Verification.service.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+  static async init(options?: VerificationServiceInitOptions): Promise<void> {
+    this.veramoAgent = await createVeramoAgent();
+  }
+
+  static async verify(
+    data: W3CVerifiablePresentation | W3CVerifiableCredential,
+    options?: { ebsiChecks?: boolean }
+  ): Promise<Result<VerificationResult>> {
+    const { ebsiChecks } = options || { ebsiChecks: false };
```
Comment (pseudobun) [inline libs/extended-verification/src/Verification.service.ts:61]: `ebsiChecks` is not used anywhere. Is this intentional?

File: libs/extended-verification/src/Verification.service.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+              // errorMessage,
+            ],
+          },
+          isExpired: errorMessage.includes('expired'),
+          isNotYetValid: errorMessage.includes('nbf'),
+        };
+
+        // Extract credentials from the presentation
+        for (const credential of presentation.verifiableCredential) {
+          credentialsToVerify.push(decodeCredentialToObject(credential));
```
Comment (pseudobun) [inline libs/extended-verification/src/Verification.service.ts:123]: I would rather use corresponding methods from the same library.
Either:
`decodeCredentialToObject` and `decodePresentationToObject` from `@veramo/utils`
or
`normalizeCredential` and `normalizePresentation` from `did-jwt-vc`

Related to lines 77 and 132 in this file.

File: libs/extended-verification/src/Verification.service.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+          if (!result.verified) {
+            verificationResult.verified = false;
+          }
+
+          const issuer =
+            typeof credential.issuer === 'string'
+              ? credential.issuer
+              : credential.issuer.id;
+
+          if (issuer.startsWith('did:ebsi:')) {
```
Comment (pseudobun) [inline libs/extended-verification/src/Verification.service.ts:158]: Maybe comment out this whole if statement and move the TODO above it.

File: libs/extended-verification/src/Verification.service.ts [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+            ebsiTrustedIssuer: false,
+          });
+        }
+
+        // TODO: Set verification to false if any of the credentials are not valid
+
+        return ResultObject.success(verificationResult);
+      }
+
+      // TODO: Test if it is JWZ format}
```
Comment (pseudobun) [inline libs/extended-verification/src/Verification.service.ts:185]: Remove } at the end because of my OCD. :)

**Review by pseudobun — CHANGES_REQUESTED** [review]

File: libs/extended-verification/src/Verification.service.ts (hunk 1 above)
Comment (martines3000) [reply libs/extended-verification/src/Verification.service.ts:60]: This is intentional, we will use this variable when we do EBSI checks.

**Review by pseudobun — APPROVED** [review]

**Review by pseudobun — APPROVED** [review]
Good job!

**Review by pseudobun — APPROVED** [review]

**Pull request merged**